core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/extensions.scala

```diff
@@ -2,4 +2,18 @@ package com.databricks.labs.remorph.parsers.intermediate
 
 trait AstExtension
 
-case class Column(name: String) extends Expression with AstExtension {}
\ No newline at end of file
+case class Column(name: String) extends Expression with AstExtension {}
+
+abstract class Predicate extends Expression with AstExtension {}
+
+case class And(left: Predicate, right: Predicate) extends Predicate {}
+case class Or(left: Predicate, right: Predicate) extends Predicate {}
+case class Not(pred: Predicate) extends Predicate {}
+
+case class Equals(left: Expression, right: Expression) extends Predicate {}
+case class NotEquals(left: Expression, right: Expression) extends Predicate {}
+case class GreaterThan(left: Expression, right: Expression) extends Predicate {}
+case class LesserThan(left: Expression, right: Expression) extends Predicate {}
+case class GreaterThanOrEqual(left: Expression, right: Expression) extends Predicate {}
+case class LesserThanOrEqual(left: Expression, right: Expression) extends Predicate {}
```

Comment:
```suggestion
case class Equals(left: Expression, right: Expression) extends Binary(left, right) {}
case class NotEquals(left: Expression, right: Expression) extends Binary(left, right) {}
case class GreaterThan(left: Expression, right: Expression) extends Binary(left, right) {}
case class LesserThan(left: Expression, right: Expression) extends Binary(left, right) {}
case class GreaterThanOrEqual(left: Expression, right: Expression) extends Binary(left, right) {}
case class LesserThanOrEqual(left: Expression, right: Expression) extends Binary(left, right) {}
```

Reply:
I agree with your `Binary`/`Unary` proposal, but I would like to also maintain the distinction between `Expression` and `Predicate`, so stay tuned for a solution that slightly differs from what you suggested.